ts/a11y/sre.ts

```diff
@@ -78,6 +78,18 @@ export namespace Sre {
 
 }
 
+/**
+ * A promise that resolves when SRE is loaded and ready, and rejects if
+ * SRE can't be loaded, or does not become ready within the timout period.
+ *
+ * @deprecated
+ */
+export const sreReady = function() {
+  return new Promise<void>((resolve, reject) =>
+    Sre.sreReady().then(() => resolve())
+      .catch((error: Error) => reject(error.message || error)));
+};
```

Comment:
Here again, I think this can be simplified.  Since Sre.sreReady() is already a promise, I think all you need is 

```sugestion
export const sreReady = ()  => Sre.sreReady();
```

Reply:
I had that at first, but then, we don't have a catch anymore. I am happy to do this, but it does change the behaviour a bit.

Reply:
Because the old `sreReady()` used polling to determine when SRE was ready, together with a timeout so that if SRE wasn't ready after a certain time, it would time out and report an error, the `catch()` was to handle that case (I didn't know much about promises then and should have simplified the usage).  The `catch()` in the current `sreReady` ends up essentially throwing the error message, so you can use

```suggestion
export const sreReady = ()  => Sre.sreReady().catch((error: Error) => {throw (error.message || error)});
```

to get the same result.

Reply:
Thought I think it would be fin to leave off the `catch()` and simply let the full error object go through rather than just the message string.

Reply:
OK, I'll leave off the catch.

Reply:
> Here again, I think this can be simplified. Since Sre.sreReady() is already a promise, I think all you need is
> 
> ```
> export const sreReady = ()  => Sre.sreReady();
> ```

Put back my original export line  `export const sreReady = Sre.sreReady;` which is even shorter...

Reply:
Just making sure that `Sre.sreReady` doesn't use `this` internally, as that would no longer work with the definition as is.  I guess `sreReady = Sre.sreReady.bind(Sre)` would do it in that case.